sockeye/vocab.py

```diff
@@ -32,23 +32,18 @@
 InverseVocab = Dict[int, str]
 
 
-def build_from_paths(paths: Union[Sequence[str], str]) -> Counter:
+def count_tokens_for_path(path: str) -> Counter:
     """
-    :param paths: List of paths to files with one sentence per line.
+    :param path: Path to file with one sentence per line.
     :return: Token counter.
     """
     with ExitStack() as stack:
-        logger.info("Building vocabulary from dataset(s): %s", paths)
-        if isinstance(paths, List):
-            files = (stack.enter_context(utils.smart_open(path, mode='rt')) for path in paths)
-        elif isinstance(paths, str):
-            files = stack.enter_context(utils.smart_open(paths, mode='rt'))
```

Comment:
Here we should have created a single element list as otherwise `chain` below iterates over characters.